packages/cdktf-cli/lib/project-execution.ts

```diff
@@ -0,0 +1,448 @@
+    },
+    states: {
+      idle: {
+        on: {
+          START: {
+            target: "synth",
+            actions: assign({
+              targetAction: (_context, event) => event.targetAction,
+              targetStack: (_context, event) => event.targetStack,
+            }),
+          },
+        },
+      },
+      synth: {
+        invoke: {
+          id: "synth",
+          src: "synthProject",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => event.data,
+            }),
+          },
+          onDone: [
+            {
+              target: "done",
+              actions: assign({
+                synthesizedStacks: (_context, event) => event.data,
+              }),
+              cond: {
+                type: "onTargetAction",
+                name: "actionIsSynth",
+                value: "synth",
+              },
+            },
+            {
+              target: "gatherOutput",
+              actions: assign({
+                synthesizedStacks: (_context, event) => event.data,
+              }),
+              cond: {
+                type: "onTargetAction",
+                name: "actionIsOutput",
+                value: "output",
+              },
+            },
+            {
+              target: "diff",
+              actions: assign({
+                synthesizedStacks: (_context, event) => event.data,
+              }),
+            },
+          ],
+        },
+      },
+      diff: {
+        invoke: {
+          id: "diff",
+          src: "diff",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => {
+                const errorMessage = event.data.stderr || event.data;
+                return `terraform plan errored with: \n${errorMessage}`;
+              },
+            }),
+          },
+          onDone: [
+            {
+              target: "done",
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              cond: {
+                type: "onTargetAction",
+                name: "actionIsDiff",
+                value: "diff",
+              },
+            },
+
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "gatherOutput",
+              cond: "planNeedsNoApply",
+            },
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "approved",
+              cond: "autoApprove",
+            },
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "waitingForApproval",
+            },
+          ],
+        },
+      },
+      waitingForApproval: {
```

Comment:
nit: This state waits for approval to either deploy or destroy, right? Can we make this more clear somehow? Do we need to? Probably not, just wanted to share my confusion 😁 Feel free to just resolve this after reading ^^